service/controller/resource/clustersecret/desired.go

```diff
@@ -48,7 +51,19 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 				if err != nil {
 					return nil, microerror.Mask(err)
 				}
+			case "VCDCluster":
+				var infraCluster capvcd.VCDCluster
+				err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{Namespace: infrastructureRef.Namespace, Name: infrastructureRef.Name}, &infraCluster)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+
+				values["global"], err = r.generateCloudDirectorConfig(ctx, infraCluster)
```

Comment:
For OpenStack we set the config under `cloudConfig` key, for VCD under `global`. Is there a specific reason / would it not be better to use the same key?

Reply:
It comes from the `values.yaml` of `cloud-provider-<provider>` apps.

https://github.com/giantswarm/cloud-provider-cloud-director-app/blob/main/helm/cloud-provider-cloud-director/values.yaml#L1

https://github.com/giantswarm/cloud-provider-openstack-app/blob/master/helm/cloud-provider-openstack-app/values.yaml#L11

We consume upstream chart in OpenStack side so the values.yaml structure is given there. @vxav Did we consume `cloud-provider-cloud-director-app` from somewhere? Can we change it like OpenStack?

Reply:
There is no upstream chart for CAPVCD so I made it myself. Cloud Config is the actual name for the creds/secret in OpenStack specifically so it doesn't make much sense to use it in VCD.
The global variable is here to avoid duplicating things and reference the parent values in the children charts.

Reply:
Also agree here, cloud config is an openstack term and doesn't make sense for other providers; customers coming from VMware won't know what it means.

Reply:
Okay. Let's keep it as it is.